src/util/strencodings.h

```diff
@@ -65,7 +65,32 @@ std::string EncodeBase32(Span<const unsigned char> input, bool pad = true);
  */
 std::string EncodeBase32(const std::string& str, bool pad = true);
 
+
+/**
+ * Splits socket address string into host and port strings.
+ *
+ * @param[in] in        The socket address string to split.
+ * @param[out] portOut  Port-portion of the input, if found.
+ * @param[out] hostOut  Host-portion of the input, if found.
+ */
+void SplitHostPort(std::string in, std::string& portOut, std::string& hostOut);
+/**
+ * Splits socket address string into host string and port value.
+ * Sanitizes port value.
+ *
+ * @param[in] in        The socket address string to split.
+ * @param[out] portOut  Port-portion of the input, if found and parsable.
+ * @param[out] hostOut  Host-portion of the input, if found.
+ */
 void SplitHostPort(std::string in, uint16_t& portOut, std::string& hostOut);
```

Comment:
I would prefer this to be more explicit about parse errors. For example:

- Return a boolean `true` if valid `false` otherwise
- Return `std::optional<uint16_t>` (or even `std:::optional<std::pair<string, uint16_t>>`)

Either is a lot clearer than "we don't assign anything if the port is invalid", which is somewhat indirect behavior.

Reply:
Changed to return bool